### PR TITLE
Add support for setting XMPP priority

### PR DESCRIPTION
**Pull request by lineprinter**

### CODE REVIEW

File: src/main/res/xml/preferences.xml [hunk 1]
```diff
@@ -16,6 +16,12 @@
             android:key="resource"
             android:summary="@string/pref_xmpp_resource_summary"
             android:title="@string/pref_xmpp_resource"/>
+        <EditTextPreference
+            android:defaultValue="0"
+            android:key="priority"
+            android:summary="Resource priority"
```
Comment (lineprinter) [inline src/main/res/xml/preferences.xml]: Fixed.

**Comment by licaon-kter** [comment]
Ok, now, this is used how exactly?

I see you have another PR in the works with editable "resource", although @iNPUTmice wanted to [move away from resources all together](https://github.com/siacs/Conversations/issues/1922#issuecomment-227787829), and this was kinda answered before (see **strb**'s comment with links: https://github.com/siacs/Conversations/issues/1937#issuecomment-230605032 )

**Comment by whitequark** [comment]
Not having support for setting XMPP priority means that Conversations is essentially unusable together with another client on a single account. For example, if Conversations uses the default priority 0, and gajim also has priority 0, and both have OTR enabled, it is impossible for the other party to initiate an OTR session with Conversations.

**Comment by iNPUTmice** [comment]
> if Conversations uses the default priority 0, and gajim also has priority 0, and both have OTR enabled, it is impossible for the other party to initiate an OTR session with Conversations.

That's not true. The initiating party should send the OTR message to a specific resource as explained in [XEP-0364 under section 4.2](http://xmpp.org/extensions/xep-0364.html#sect-idp662768).

**Comment by whitequark** [comment]
@iNPUTmice That doesn't actually work with Pidgin.

**Comment by iNPUTmice** [comment]
So your solution to deal with a broken and outdated client is to introduce
unnecessary code in Conversations instead of fixing the broken client?

**Comment by benediktg** [comment]
> For example, if Conversations uses the default priority 0, and gajim also has priority 0, and both have OTR enabled

Why would one want to use OTR with gajim and Conversations when there's also OMEMO?

**Comment by whitequark** [comment]
> So your solution to deal with a broken and outdated client is to introduce
> unnecessary code in Conversations instead of fixing the broken client?

Interoperability through features standardized in protocol that solve a real-world problem is hardly "unnecessary". Maybe if you want to ediorialize instead of helping people.

> Why would one want to use OTR with gajim and Conversations when there's also OMEMO?

gajim and Conversations would be on the same account, and the other party doesn't necessarily have OMEMO. (Although, wouldn't OMEMO have the same issue anyhow?)

**Comment by benediktg** [comment]
> gajim and Conversations would be on the same account, and the other party doesn't necessarily have OMEMO.

gajim and Conversations are the most recommendable XMPP clients anyway (at the moment).

> wouldn't OMEMO have the same issue anyhow?

OMEMO does not have this issue because it plays (or it should) play well with Message Carbons so that a copy of the message gets sent to every client of your chat partner of which you have trusted it's OMEMO public key.

**Comment by whitequark** [comment]
> OMEMO does not have this issue because it plays (or it should) play well with Message Carbons so that a copy of the message gets sent to every client of your chat partner of which you have trusted it's OMEMO public key.

OK, I see. Arguably this would be an issue with OTR then, more so than Conversations.

**Comment by iNPUTmice** [comment]
Priority is an outdated concept; Carbons can fully replace the server side guessing nowadays.

**Pull request closed without merging**